### PR TITLE
query/ui: Added support for specifying max source resolution.

### PR DESCRIPTION
**Pull request by Bplotka**
Related to https://github.com/improbable-eng/thanos/issues/310

- Added queryRange maxSourceResolution parameter.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by Bplotka** [comment]
image

File: pkg/query/api/v1.go [hunk 1]
```diff
@@ -265,14 +265,27 @@ func (api *API) queryRange(r *http.Request) (interface{}, []error, *apiError) {
...
+		return nil, nil, &apiError{errorBadData, errors.Wrap(err, "param step")}
 	}
 
 	if step <= 0 {
 		err := errors.New("zero or negative query resolution step widths are not accepted. Try a positive integer")
 		return nil, nil, &apiError{errorBadData, err}
 	}
 
+	maxSourceResolution := step / 5 // By default fit at least 5 samples between steps.
+	if val := r.FormValue("max_source_res"); val != "" {
```
Comment (stefan-improbable) [inline pkg/query/api/v1.go]: Nit: can we name this `max_source_resolution`? I have a bad experience with naming the same thing with 2 slightly different names, it increases confusion level a bit.

Comment (Bplotka) [reply pkg/query/api/v1.go]: makes sense!

File: pkg/query/querier.go [hunk 2]
```diff
@@ -21,43 +22,47 @@ type PartialErrReporter func(error)
...
 	}
 }
 
 type queryable struct {
-	logger           log.Logger
-	replicaLabel     string
-	proxy            storepb.StoreServer
-	deduplicate      bool
-	partialErrReport PartialErrReporter
+	logger              log.Logger
```
[the remark this answers is not in the document]
Comment (Bplotka) [reply pkg/query/querier.go]: yea, let's leave it. Basically there are no obvious things like `Querier` only context etc.

**Pull request merged**